examples/prom/grad_div_global_rom.cpp

```diff
@@ -0,0 +1,569 @@
+    //    corresponding to fespace. Initialize x by projecting the exact
+    //    solution. Note that only values from the boundary faces will be used
+    //    when eliminating the non-homogeneous boundary condition to modify the
+    //    r.h.s. vector b.
+    ParGridFunction x(&fespace);
+    VectorFunctionCoefficient F(sdim, F_exact);
+    x.ProjectCoefficient(F);
+
+    // 14. Set up the parallel bilinear form corresponding to the H(div)
+    //     diffusion operator grad alpha div + beta I, by adding the div-div and
+    //     the mass domain integrators.
+    ConstantCoefficient alpha(1.0);
+    ConstantCoefficient beta(1.0);
+    ParBilinearForm a(&fespace);
+    if (pa) {
+        a.SetAssemblyLevel(AssemblyLevel::PARTIAL);
+    }
+    a.AddDomainIntegrator(new DivDivIntegrator(alpha));
+    a.AddDomainIntegrator(new VectorFEMassIntegrator(beta));
+
+    // 15. Assemble the parallel bilinear form and the corresponding linear
+    //     system, applying any necessary transformations such as: parallel
+    //     assembly, eliminating boundary conditions, applying conforming
+    //     constraints for non-conforming AMR, static condensation,
+    //     hybridization, etc.
+    FiniteElementCollection *hfec = NULL;
+    ParFiniteElementSpace *hfes = NULL;
+    if (static_cond)
+    {
+        a.EnableStaticCondensation();
+    }
+    else if (hybridization)
+    {
+        hfec = new DG_Interface_FECollection(order-1, dim);
+        hfes = new ParFiniteElementSpace(&pmesh, hfec);
+        a.EnableHybridization(hfes, new NormalTraceJumpIntegrator(),
+                              ess_tdof_list);
+    }
+    a.Assemble();
+
+    OperatorPtr A;
+    Vector B, X;
+    a.FormLinearSystem(ess_tdof_list, x, *b, A, X, B);
+    assembleTimer.Stop();
+
+    if (myid == 0 && !pa)
+    {
+        cout << "Size of linear system: "
+             << A.As<HypreParMatrix>()->GetGlobalNumRows() << endl;
+    }
+
+    // 16. The offline phase
+    if(fom || offline)
+    {
+        // 17. Define and apply a parallel PCG solver for A X = B with the 2D AMS or
+        //     the 3D ADS preconditioners from hypre. If using hybridization, the
+        //     system is preconditioned with hypre's BoomerAMG. In the partial
+        //     assembly case, use Jacobi preconditioning.
+        Solver *prec = NULL;
+        CGSolver *pcg = new CGSolver(MPI_COMM_WORLD);
+        pcg->SetOperator(*A);
+        pcg->SetRelTol(1e-12);
+        pcg->SetMaxIter(2000);
+        pcg->SetPrintLevel(1);
+        if (hybridization) {
+            prec = new HypreBoomerAMG(*A.As<HypreParMatrix>());
+        }
+        else if (pa) {
+            prec = new OperatorJacobiSmoother(a, ess_tdof_list);
+        }
+        else
+        {
+            ParFiniteElementSpace *prec_fespace =
+                (a.StaticCondensationIsEnabled() ? a.SCParFESpace() : &fespace);
+            if (dim == 2)   {
```

Comment:
Not a big deal, but the spaces before `{` here and in the `else` line are unusual.